### ISSUE TITLE
Encrypt with python?

### ISSUE THREAD

**Issue by LeoMyMe** [FirehawkV21/RMMVCookTool#81]
(Optional and Optional choice in RMMVCookTool)How about mixed encrypt some files with python(.pyd) and pyinstaller?

**Comment by FirehawkV21** [comment]
Could you please elaborate on what you'd like to do?

**Comment by FirehawkV21** [comment]
The Cook Tool compiles the JS files to binaries that NW.js understands. This is used for protecting the game's source code. Due to the compiler that the tool uses (nwjc.exe, included with the NW.js SDK), it does make the source code specific to the version of NW.js and the platform where the compiler ran (so, if you compiled it for Windows, the game's code will only run on Windows).

**Comment by LeoMyMe** [comment]
In my mind,I think it has two parts.First part is python part and second part is game(js) part.
(First think)Python part has the key which is encrypted.Game part has the way to decrypt.When player clicks the exe file(python),python part start connection with game part.Then python part send the key to game part.At last,the game part decrypt the key and start the game(if True).
(Second think)Same as first think.But python part has the way to decrypt.Game part has the key which is encrypted.
But these ideas are not mature.

**Comment by LeoMyMe** [comment]
In my mind,I think it has two parts.First part is python part and second part is game(js) part.
(First think)Python part has the key which is encrypted.Game part has the way to decrypt.When player clicks the exe file(python),python part start connection with game part.Then python part send the key to game part.At last,the game part decrypt the key and start the game(if True).
(Second think)Same as first think.But python part has the way to decrypt.Game part has the key which is encrypted.
But these ideas are not mature.

**Comment by RIKDXHQ11** [comment]
> The Cook Tool compiles the JS files to binaries that NW.js understands. This is used for protecting the game's source code. Due to the compiler that the tool uses (nwjc.exe, included with the NW.js SDK), it does make the source code specific to the version of NW.js and the platform where the compiler ran (so, if you compiled it for Windows, the game's code will only run on Windows).

Can I compile them on Mac or Linux and have them run on these platforms?

**Comment by FirehawkV21** [comment]
> > The Cook Tool compiles the JS files to binaries that NW.js understands. This is used for protecting the game's source code. Due to the compiler that the tool uses (nwjc.exe, included with the NW.js SDK), it does make the source code specific to the version of NW.js and the platform where the compiler ran (so, if you compiled it for Windows, the game's code will only run on Windows).
> 
> Can I compile them on Mac or Linux and have them run on these platforms?

Yes. However, you'd need to use the Mac/Linux version of the NW.js SDK (alongside a compiled version of Cook Tool for said OS). In my case, I used WSL (Windows Subsystem for Linux), in order to create a Linux binary.

**Comment by RIKDXHQ11** [comment]
> > > The Cook Tool compiles the JS files to binaries that NW.js understands. This is used for protecting the game's source code. Due to the compiler that the tool uses (nwjc.exe, included with the NW.js SDK), it does make the source code specific to the version of NW.js and the platform where the compiler ran (so, if you compiled it for Windows, the game's code will only run on Windows).
> > 
> > 
> > Can I compile them on Mac or Linux and have them run on these platforms?
> 
> Yes. However, you'd need to use the Mac/Linux version of the NW.js SDK (alongside a compiled version of Cook Tool for said OS). In my case, I used WSL (Windows Subsystem for Linux), in order to create a Linux binary.



Okay, I See.
At the same time, I found that pixel game maker mv and RMMV are the same company, and this game engine also has many similarities with RMMV, using JavaScript as the scripting language. Although it is very good, it is difficult to encrypt because it is an open folder. If it can be compiled, it will be usable. I am not sure if your tool can also be applied to it. If not, I hope you can add support for it. Many developers who use it will be very happy. Thank you.
